client/client.go

```diff
@@ -251,6 +251,8 @@ func defaultHTTPClient(hostURL *url.URL) (*http.Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	transport.MaxIdleConns = 6
+	transport.IdleConnTimeout = 30 * time.Second
```

Comment:
Perhaps worth to add a comment linking to the issue here

Reply:
Curious; would setting these on the transport we _pass_ to `ConfigureTransport` work? i.e., I don't _think_ it overwrites properties it doesn't set? But perhaps it would make it more future-proof to consider these defaults we set (but can be overwritten by `ConfigureTransport`;

i.e.;

```go
	transport := &http.Transport{
		MaxIdleConns:    6,
		IdleConnTimeout: 30 * time.Second,
	}
	err := sockets.ConfigureTransport(transport, hostURL.Scheme, hostURL.Host)
```

Reply:
> Curious; would setting these on the transport we pass to ConfigureTransport work?

It would work, sure, I'm okay with doing that. I'd rather not bikeshed too much on this change though, I want to take a proper look at go-connections afterwards and see if we can some changes there and then we can revisit this, this was just meant as a quick port of tonis's patch to address the immediate issue.